### PR TITLE
Structure migration

### PR DESCRIPTION
**Pull request by andrzej-casper**
## :fireworks: Live demo

https://bespoke-pika-68a036.netlify.app/

## :fire: What is happening?

Let me briefly give a bit of transparency to community and people in Casper. Current documentation is undergoing heavy changes, as soon (end of March?) it will be seamlessly embedded into an upcoming *Developer Portal* (with fancy UI, nice UX). This means some content will be moved around within *Docs*, or across *Portal* and *Docs* layers. We also expect to refactor existing pages and create new ones. Obviously transition process is difficult, however we will try to make it least disruptive as possible.

## :bookmark_tabs: New structure

New structure for *Docs* layer was prepared by external company, that did a lot of research (e.g. analyzed existing content, medium articles, compared with other blockchains). We improved it a little bit, and now we have the basic idea how content will be distributed. We treat it rather as a guideline, not strictly the final version.

## :checkered_flag: Objectives

This PR is **first step** toward the new structure. For now we entirely focus on moving pages, and preparing empty slots for new ones... but no content is changed or deleted! We also have to update existing links, adjust *sidebar* and *navbar* configs.

**Important:** We expect that logic flow can be broken now, and some sections will need refactor, new titles etc. That is the deal, when we agreed to work on existing repository, rather than start the work from scratch.

## :speech_balloon: Content of this PR

I will describe each commit, as it will be useful for review purposes.

**1. Adjust files to new structure**

Pages were moved automatically according to [migration CSV](https://gist.github.com/andrzej-casper/2ef0408f5535c452ae8c5c8848908953). Simply a lot of renames, but in commit you will see one additional change, which is _slug_ update - I had to do it to avoid conflicts. For review I would advise to open [live demo](https://bespoke-pika-68a036.netlify.app/) and see visually what changed.

**2. Update links after migration**

Above mentioned CSV file was also used to automatically update links. There is no reason to go through changes manually (maybe pick few randomly) - just execute `yarn build` to make sure all links are validated by Docusaurus.

**3. Adjust navbar to new structure**

We basically want 4 sections: _Concepts_, _Developers_, _Operators_ and _Resources_. However, before *Portal* layer is ready, we had to store existing content somewhere **temporarily**. That is why we have additional _User_ section, and we leaved _FAQ_, _Chat_.

**4. Adjust sidebar to new structure**

Manual change, that involved going through existing directory structure. If there would have been something wrong (resource with no matching file), Docusaurus would report it during build process. You might be only worried about files with no resources defined - this can be validated manually by going through all `*.md` files and looking for corresponding resource.

All resources for new content are commented out, to avoid empty pages visible at https://docs.casperlabs.io/.

**5. Add missing indexes for new sections**

Since there are new sections/subsections defined - e.g. _Concepts_ - we have to create `index.md` for them. Otherwise links like `/concepts/` would be somewhat broken. For now I filled them with table of content, but we can expect they will be updated later.

**6. Prepare pages for new content**

This step was optional, but I think it will be easier for people to have pre-created files, when they work on new content.

**7. Update link in tutorial template**

Small change to update link to counter tutorial in GitHub template.

**8. Update links in pipeline checks**

There are checks in CI/CD that depends on URLs like `/operators/create`, but since content was moved I had to update those links.

## :gear: Future work

Although we have successfully relocated **123** pages, we encountered some difficulty with the remaining **5** pages, specifically:
- dapp-dev-guide/json-rpc
- dapp-dev-guide/building-dapps/dapp
- ~~dapp-dev-guide/building-dapps/calling-contracts~~ **Update:** placed under _Developers_ / _Writing On-Chain Code_
- dapp-dev-guide/building-dapps/setup-nctl
- dapp-dev-guide/building-dapps/nctl-test

AFAIR, it seems like the stuff inside those files might be repeated elsewhere. We didn't delete them, but we'll figure out what to do with the content in the future.

### CODE REVIEW

**Comment by andrzej-casper** [comment]
As requested for this PR during today's mob review:

1. How To's index removed.
2. dapp-dev-guide/building-dapps/calling-contracts placed under "Writing On-Chain Code".
3. Announcement bar added:

announcement

@darthsiroftardis @ipopescu Let's light this candle! :rocket:

File: source/docs/casper/developers/table-of-content.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+- [Prerequisites](./prerequisites.md): setup needed for various workflows
+- [Querying a Casper Network](../resources/tutorials/beginner/querying-network.md): querying information on a Casper network
+- [Two-Party Multi-Signature Deploys](../resources/tutorials/advanced/two-party-multi-sig.md): setting up an account for two-party signature deploys
+- [Transferring Tokens using the Command-line](./cli/transfers/index.md): transferring tokens from one account to another using the command-line
+   - [Transferring Tokens using Direct Transfer](./cli/transfers/direct-token-transfer.md)
+   - [Transferring Tokens using Multi-sig Account](./cli/transfers/multisig-deploy-transfer.md)
+   - [Verifying a Transfer](./cli/transfers/verify-transfer.md)
+- [Delegating Tokens](./cli/delegate.md): delegating tokens to a Validator on a Casper network
+- [Undelegating Tokens](./cli/undelegate.md): undelegating tokens from a validator on a Casper network
+- [Using the Casper Fungible Token Contract](https://github.com/casper-network/erc20-guide-extraction/): using the Casper fungible token contract to understand the fungible token transfer workflow
```
Comment (ipopescu) [inline source/docs/casper/developers/table-of-content.md:16]: The content listed here doesn’t match the new structure. Please update.

File: source/docs/casper/operators/table-of-content.md [hunk 2]
```diff
@@ -0,0 +1,6 @@
+import useBaseUrl from '@docusaurus/useBaseUrl';
+
+# Table of Contents
+
+- [Set Up a Private Casper Network](./setup-network/create-private.md): a step-by-step guide to establishing and configuring a private Casper network
+- [How to Stage Files for a New Network](./setup-network/staging-files-for-new-network.md): a guide to hosting protocol files for a new Casper network
```
Comment (ipopescu) [inline source/docs/casper/operators/table-of-content.md:6]: The content listed here doesn’t match the new structure. Please update.

**Review by ipopescu — CHANGES_REQUESTED** [review]
Thanks @andrzej-casper for the thorough work and explanation.
I have left several requests before merging this PR to the live site.

Also, 3 pages with “table of contents” do not list the contents correctly according to the new structure. I wasn’t able to leave a comment for the users/index.md page that needs to be updated to match the new structure.
- /developers/table-of-content/ 
- /operators/table-of-content/ 
- /users/ 

TODOs post-merge. FYI @Adrian-Wrona.

As discussed, we need to review and revise the order in the left navigation bar. Also, several inconsistencies exist in the new headings introduced (Introduction, Overview, Table of Contents). Ideally, this would all be sorted before the content is pushed live, but we can address it asap afterward.

Also, I am not sure why these are rendered with the incorrect case in the new structure:
	Interacting with the Blockchain Using CLI
	Becoming a Validator
	Setting Up a Network
It is not a PR blocker but needs to be fixed after this PR. 

I found a few inconsistent titles that need to be updated later (from the old content):
Setting Up a Private Casper Network to match the other titles
Block Explorers on Casper Network => Block Explorers for Casper Networks

I also saw this incorrectly rendered note while reviewing, but I don’t want to update until this PR merges.

::note
Please make sure you are running this tool as the same user that owns $DATA_DIR. Otherwise, you may receive a permission denied error.
:::

File: config/sidebar.config.js [hunk 3]
```diff
@@ -1,253 +1,332 @@
...
+        "users/delegate-ui",
+        "users/undelegate-ui",
+        "users/ledger-setup",
+        "users/token-transfer",
+        "users/funding-from-exchanges",
+        "users/testnet-faucet",
         {
             type: "category",
-            label: "Runtime Economics",
+            label: "Why Build on Casper",
```
[the remark this answers is not in the document]
Comment (ipopescu) [reply config/sidebar.config.js:285]: This needs to be done as part of this PR.

File: source/docs/casper/operators/table-of-content.md (hunk 2 above)
Comment (ipopescu) [reply source/docs/casper/operators/table-of-content.md:6]: Please do the same for the users table of contents.

File: config/navbar.config.js [hunk 4]
```diff
@@ -35,27 +35,15 @@ module.exports = {
...
-            to: `${routePrefix}/staking`,
-            activeBasePath: `${routePrefix}/staking`,
-            label: "Staking",
-            position: "left",
-        },
-        {
-            to: `${routePrefix}/glossary`,
-            activeBasePath: `${routePrefix}/glossary`,
-            label: "Glossary",
+            to: `${routePrefix}/users`,
```
[the remark this answers is not in the document]
Comment (andrzej-casper) [reply config/navbar.config.js:38]: Fixed in https://github.com/casper-network/docs/pull/912/commits/0d21967d94cd7253ed79cf8577721282a6e8cfaa.

File: config/sidebar.config.js [hunk 5]
```diff
@@ -1,253 +1,332 @@
...
+        //        //                    collapsed: true,
+        //        //                    items: [
+        //        //                        "dapp-dev-guide/tutorials/multi-sig/index",
+        //        //                        "dapp-dev-guide/tutorials/multi-sig/concepts",
+        //        //                        "dapp-dev-guide/tutorials/multi-sig/contract",
+        //        //                        "dapp-dev-guide/tutorials/multi-sig/client",
+        //        //                        "dapp-dev-guide/tutorials/multi-sig/example",
+        //        //                        "dapp-dev-guide/tutorials/multi-sig/additional",
+        //        //                   ],
+        //        //                },
```
[the remark this answers is not in the document]
Comment (andrzej-casper) [reply config/sidebar.config.js:328]: Task tracked internally as `CAS-64`.

File: source/docs/casper/developers/cli/index.md [hunk 6]
```diff
@@ -0,0 +1,8 @@
+# Overview
```
[the remark this answers is not in the document]
Comment (andrzej-casper) [reply source/docs/casper/developers/cli/index.md:1]: Further discussion is required - task tracked internally as `CAS-65`.

File: source/docs/casper/resources/tutorials/index.md [hunk 7]
```diff
@@ -0,0 +1,4 @@
+# Overview
+
```
[the remark this answers is not in the document]
Comment (andrzej-casper) [reply source/docs/casper/resources/tutorials/index.md:2]: Task tracked internally as `CAS-60`.

File: config/sidebar.config.js [hunk 8]
```diff
@@ -1,253 +1,332 @@
...
+                //"developers/dapps/using-casper-signer", // NEW CONTENT WILL BE HERE
+                //"developers/dapps/callstack-based", // NEW CONTENT WILL BE HERE
+                //"developers/dapps/explanation-session-and-contract", // NEW CONTENT WILL BE HERE
+                "developers/dapps/monitor-and-consume-events",
             ],
         },
         {
             type: "category",
-            label: "Writing On-Chain Code",
+            label: "Interacting with the Blockchain using CLI",
```
[the remark this answers is not in the document]
Comment (andrzej-casper) [reply config/sidebar.config.js:152]: This is the default behavior of Docusaurus - it uses uppercase for all collapsible items. This applies to https://docs.casperlabs.io/ as well:

image

If you think case should not be transformed, please create relevant issue and let me know. I can provide simple CSS workaround for it :wink:.

File: config/announcement.config.js [hunk 9]
```diff
@@ -1,4 +1,7 @@
 module.exports = {
-    // id: "supportus",
-    // content: "REPLACE_WITH_YOUR_ANNOUNCEMENT!",
+    id: "announcement",
+    content: "Documentation is undergoing changes for our new <b>Developer Portal</b> launching in Q1. Please excuse any disruptions.",
```
[the remark this answers is not in the document]
Comment (andrzej-casper) [reply config/announcement.config.js]: Updated.

File: config/sidebar.config.js [hunk 10]
```diff
@@ -1,253 +1,332 @@
...
+                "developers/json-rpc/types_chain",
+                "developers/json-rpc/types_cl",
             ],
         },
-        "dapp-dev-guide/list-cspr",
-        "dapp-dev-guide/setup",
         {
             type: "category",
-            label: "Building dApps",
+            label: "Build dApps",
```
[the remark this answers is not in the document]
Comment (andrzej-casper) [reply config/sidebar.config.js]: Suggestion approved.

File: config/sidebar.config.js [hunk 11]
```diff
@@ -1,253 +1,332 @@
...
+                "developers/cli/undelegate",
+                "developers/cli/installing-contracts",
+                "developers/cli/calling-contracts",
+                "developers/cli/execution-error-codes",
             ],
         },
+    ],
+    operators: [
+        "operators/index",
+        "operators/table-of-content",
```
[the remark this answers is not in the document]
Comment (andrzej-casper) [reply config/sidebar.config.js]: Fixed in https://github.com/casper-network/docs/pull/912/commits/00c0e8931de8b0f6881587fd89c1212dce8ebdde.

File: config/sidebar.config.js [hunk 12]
```diff
@@ -1,253 +1,332 @@
...
-                        "dapp-dev-guide/tutorials/counter/commands",
-                        "dapp-dev-guide/tutorials/counter/walkthrough",
+                        "resources/tutorials/beginner/index",
+                        "resources/tutorials/beginner/getting-started-tutorial",
+                        "resources/tutorials/beginner/use-javascript-sdk",
+                        "resources/tutorials/beginner/querying-network",
+                        "resources/tutorials/beginner/upgrade-contract",
+                        {
+                            type: "category",
+                            label: "Build Counter on the Testnet",
```
[the remark this answers is not in the document]
Comment (andrzej-casper) [reply config/sidebar.config.js]: Original title restored in https://github.com/casper-network/docs/pull/912/commits/f79f01fbadd730252b0e29bd674f946b234fa47d.

File: config/sidebar.config.js [hunk 13]
```diff
@@ -1,253 +1,332 @@
...
+                            items: [
+                                "resources/tutorials/beginner/counter-testnet/index",
+                                "resources/tutorials/beginner/counter-testnet/overview",
+                                "resources/tutorials/beginner/counter-testnet/commands",
+                                "resources/tutorials/beginner/counter-testnet/walkthrough",
+                            ],
+                        },
+                        {
+                            type: "category",
+                            label: "Build Counter on an NCTL Network",
```
[the remark this answers is not in the document]
Comment (andrzej-casper) [reply config/sidebar.config.js]: Original title restored in https://github.com/casper-network/docs/pull/912/commits/f79f01fbadd730252b0e29bd674f946b234fa47d.

File: config/sidebar.config.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (andrzej-casper) [reply config/sidebar.config.js]: Moved in https://github.com/casper-network/docs/pull/912/commits/291d05016363b52ee3f4dfa734f09fd55475acee.

File: source/docs/casper/developers/table-of-content.md (hunk 1 above)
Comment (andrzej-casper) [reply source/docs/casper/developers/table-of-content.md]: Done in https://github.com/casper-network/docs/pull/912/commits/e42c0c5239cc3cc2a6f7f68e6716b2e46326cd18.

File: source/docs/casper/operators/table-of-content.md (hunk 2 above)
Comment (andrzej-casper) [reply source/docs/casper/operators/table-of-content.md]: Done in https://github.com/casper-network/docs/pull/912/commits/e42c0c5239cc3cc2a6f7f68e6716b2e46326cd18.

**Comment by andrzej-casper** [comment]
@ipopescu All changes requested for this PR were addressed, branch rebased to latest dev.

:fireworks: New live demo: https://bucolic-zuccutto-8f268c.netlify.app/.

File: config/sidebar.config.js (hunk 8 above)
[the remark this answers is not in the document]
Comment (ipopescu) [reply config/sidebar.config.js:152]: Opened issue https://github.com/casper-network/docs/issues/933 to track this.

File: source/docs/casper/developers/table-of-content.md (hunk 1 above)
Comment (ipopescu) [reply source/docs/casper/developers/table-of-content.md]: I submitted additional commits regarding this.

File: source/docs/casper/operators/table-of-content.md (hunk 2 above)
Comment (ipopescu) [reply source/docs/casper/operators/table-of-content.md]: I submitted additional commits regarding this.

**Comment by darthsiroftardis** [comment]
My only concern currently is that we need to surface the serialization standard to a higher level. If we can figure that, I can approve the PR

**Comment by ipopescu** [comment]
As discussed, we won't standup a feature branch for this PR but a staging environment.

**Comment by andrzej-casper** [comment]
@ipopescu I went through your commits, great work! Here is the latest live demo that includes your changes: https://deft-croissant-d02536.netlify.app/.

**Comment by andrzej-casper** [comment]
@darthsiroftardis What about moving it one layer up?

image

To be honest, I would prefer to resolve those small issues later. The purpose of this PR was to get prototype of new structure **ASAP**, so we can avoid double work - until then we are forced to work on [doc](https://github.com/casper-network/docs-new). Every change on your `dev` branch will cause conflicts to resolve here, so in some way it is also blocker for you.

**Comment by andrzej-casper** [comment]
@ipopescu @darthsiroftardis One additional question regarding _staging server_. Is the idea here to setup CI/CD deploying **every** branch in this repository, to have live preview out of the box?

**Comment by andrzej-casper** [comment]
@darthsiroftardis I moved serialization page as suggested.

**Review by darthsiroftardis — APPROVED** [review]
Approved, however, subsequent PRs should be targeting the feat branch

**Comment by andrzej-casper** [comment]
> Approved, however, subsequent PRs should be targeting the feat branch

@darthsiroftardis I see no reason in using feature branch. We need to have discussion, because it seems we are going in circles.

File: config/sidebar.config.js (hunk 8 above)
[the remark this answers is not in the document]
Comment (andrzej-casper) [reply config/sidebar.config.js:152]: Fixed in https://github.com/casper-network/docs/pull/960.

File: config/sidebar.config.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (andrzej-casper) [reply config/sidebar.config.js:331]: Fixed in https://github.com/casper-network/docs/pull/958.

**Pull request merged**